app/src/main/java/com/github/steroidteam/todolist/user/User.java

```diff
@@ -0,0 +1,47 @@
+package com.github.steroidteam.todolist.user;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+
+public interface User {
+    /* Trivial method to have here */
+    boolean isLoggedIn();
+
+    /* Unique user identifier */
+    String getUserId() throws UserLoginException;
+
+    /* User profile */
+    String getDisplayName() throws UserLoginException;
+    String getEmail() throws UserLoginException;
+    boolean isEmailVerified() throws UserLoginException;
+
+    /* Actions on current user */
+    void updateDisplayName(String newDisplayName, OnCompleteListener<Void> onComplete)
+            throws UserLoginException;
+    void updateEmail(String newEmail, OnCompleteListener<Void> onComplete)
+            throws UserLoginException;
+    void sendEmailVerification(OnCompleteListener<Void> onComplete)
+            throws UserLoginException;
+    void setUserPassword(String newPassword, OnCompleteListener<Void> onComplete)
+            throws UserLoginException;
+    void sendPasswordResetEmail(OnCompleteListener<Void> onComplete)
+            throws UserLoginException;
+    void deleteUser(OnCompleteListener<Void> onComplete)
+            throws UserLoginException;
+    void reauthenticate(OnCompleteListener<Void> onComplete);
```

Comment:
Maybe this method could also throw a `UserLoginException`

Reply:
Why would we want such a method?

I guess that if the user's token has expired they will have to log back in to get a fresh one, and that is basically running the whole authentication flow as if they were logged out.